### PR TITLE
feat: Add JSON schema for specs

### PR DESCRIPTION
**Pull request by candiduslynx**
Required for file-based plugins (file, s3, etc).

### CODE REVIEW

**Comment by candiduslynx** [comment]
Refactoring is extracted into #344

File: schema.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+              "pattern": "^.$",
+              "description": "Character that will be used as the delimiter.",
+              "default": ","
+            }
+          },
+          "additionalProperties": false,
+          "type": "object",
+          "description": "CloudQuery CSV file output spec."
+        },
+        "JSONSpec": {
```
Comment (erezrokah) [inline schema.json:27]: The spec for `JSON` and `Parquet` is an empty object?

File: schema.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            }
+          ]
+        },
+        "compression": {
+          "type": "string",
+          "enum": [
+            "",
+            "gzip"
+          ],
+          "description": "Compression type.\nEmpty or missing stands for no compression."
```
Comment (erezrokah) [inline schema.json:130]: Should this be `empty for no compression`? What does `missing stands` mean?

**Review by erezrokah — COMMENTED** [review]
Looks good, have a few questions on the schema

File: schema.json (hunk 1 above)
Comment (candiduslynx) [reply schema.json:27]: yes

File: schema.json (hunk 2 above)
Comment (candiduslynx) [reply schema.json:130]: * empty: `"compression":""`
* missing: just missing from the spec

**Review by erezrokah — APPROVED** [review]

**Pull request merged**